### DISCUSSION TITLE
Get previous/next data

### DISCUSSION THREAD

**Discussion by ragokan** [prisma/prisma#12759] [Q&A] [answered]
Hello, I want to get previous/next data according to the id and to do that I don't want to get all data from the database, just the next or previous one. 

![image](https://user-images.githubusercontent.com/63511519/162634425-1648b00a-9641-4293-9188-5974221f5566.png)
The reason I need that is; this is our application's update page. When we press the next/previous, we want to go to that data. Because some of the data might be deleted, the previous number for id is not working. 
Also we have more than 10k records for this model, so it takes so much time to fetch all of them.

I tried doing something like this:
```ts
prisma.message.findFirst({
      where: {
        id: {
          gt: 20,
        },
      },
    });
```
    
This works for the *next* one, but not for the *previous* one, because it will show the first matching. I can't also do get 19 because it might be deleted.

**Comment by pantharshit00** [comment, marked as answer]
Hello @ragokan , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

You can use cursor based pagination here in order to paginate across your data. Read more about it here: https://www.prisma.io/docs/concepts/components/prisma-client/pagination

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
